### PR TITLE
refat(ocp): add feature flag to ocp workloads

### PR DESCRIPTION
**Pull request by nicolearagao**
To streamline reporting and enhance flexibility, this commit introduces a feature flag for OCP workloads. By default, the feature flag is disabled, allowing us to control the information it provides.

This refactor is part of [[DISCOVERY-251]](https://issues.redhat.com/browse/DISCOVERY-251)

### CODE REVIEW

File: quipucords/tests/scanner/openshift/test_inspect.py [hunk 1]
```diff
@@ -115,6 +118,14 @@ def cluster_err(error):
     return OCPCluster(uuid="yet-another-not-uuid", errors={"some-error": error.dict()})
 
 
+@pytest.fixture
+def use_dummy_feature_flag(settings):
```
Comment (bruno-fs) [inline quipucords/tests/scanner/openshift/test_inspect.py:122]: I think the fixture name should make it clear that workloads is enabled.
maybe `workloads_enabled`?

**Review by infinitewarp — APPROVED** [review]

File: quipucords/tests/scanner/openshift/test_inspect.py [hunk 2]
```diff
@@ -250,3 +257,45 @@ def test_inspect_errors_on_extra_cluster_facts(
...
+
+
+@pytest.mark.django_db
+def test_inspect_with_disabled_workloads(  # noqa: PLR0913
+    mocker, scan_task: ScanTask, cluster, node_ok, operators
+):
+    """Test connecting to OpenShift host with workloads default behavior."""
+    mocker.patch.object(OpenShiftApi, "retrieve_cluster", return_value=cluster)
+    mocker.patch.object(OpenShiftApi, "retrieve_nodes", return_value=[node_ok])
+    mocker.patch.object(OpenShiftApi, "retrieve_operators", return_value=operators)
```
Comment (bruno-fs) [inline quipucords/tests/scanner/openshift/test_inspect.py:293]: could you also mock "retrieve_operators" method and check it was not called after execute_task is executed?

**Review by bruno-fs — COMMENTED** [review]
Looks pretty solid! I've mad some small suggestions

File: quipucords/tests/scanner/openshift/test_inspect.py (hunk 2 above)
Comment (nicolearagao) [reply quipucords/tests/scanner/openshift/test_inspect.py:293]: retrieve_workloads u mean? sure thing, I will add that check.

File: quipucords/tests/scanner/openshift/test_inspect.py (hunk 1 above)
Comment (nicolearagao) [reply quipucords/tests/scanner/openshift/test_inspect.py:122]: Good idea!

File: quipucords/tests/scanner/openshift/test_inspect.py (hunk 2 above)
Comment (bruno-fs) [reply quipucords/tests/scanner/openshift/test_inspect.py:293]: yep, that haha

**Review by bruno-fs — APPROVED** [review]

**Pull request merged**